cmd/os_unix.go

```diff
@@ -86,6 +86,40 @@ func osMkdirAll(dirPath string, perm os.FileMode) error {
 
 	return nil
 }
+func panOSMkdirAll(dirPath string, perm os.FileMode, ownerID, groupID int) error {
+	// Fast path: if we can tell whether path is a directory or file, stop with success or error.
+	err := Access(dirPath)
+	if err == nil {
+		return nil
+	}
+	if !osIsNotExist(err) {
+		return &os.PathError{Op: "mkdir", Path: dirPath, Err: err}
+	}
+
+	// Slow path: make sure parent exists and then call Mkdir for path.
+	i := len(dirPath)
+	for i > 0 && os.IsPathSeparator(dirPath[i-1]) { // Skip trailing path separator.
+		i--
+	}
+
+	j := i
+	for j > 0 && !os.IsPathSeparator(dirPath[j-1]) { // Scan backward over element.
+		j--
+	}
+
+	if j > 1 {
+		// Create parent.
+		if err = osMkdirAll(dirPath[:j-1], perm); err != nil {
```

Comment:
Should it be a call to panOSMkdirAll instead? Or it seems we apply owner/group only to last element of path.

Reply:
Fixed